ledger-core/virtual/execute/execute.go

```diff
@@ -388,15 +391,97 @@ func (s *SMExecute) stepDeduplicateUsingPendingsTable(ctx smachine.ExecutionCont
 	}
 
 	if isDuplicate {
-		// TODO: do we have result? if we have result then we should resend result
-		ctx.Log().Warn("duplicate found as pending request")
-		return ctx.Stop()
+		if isActive {
+			ctx.Log().Warn("duplicate found as pending request")
+			return ctx.Stop()
+		}
 	}
 
 	if ctx.AcquireForThisStep(pendingListFilled).IsNotPassed() {
```

Comment:
in your PR you also need to drop pendingListFilled semaphores

Reply:
with SMAwaitTableFill ?

Reply:
yep. In deduplication with VE(P-1) there is no need to wait for full pending table